platform/FileHandle.h

```diff
@@ -152,8 +162,106 @@ class FileHandle {
      */
     MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileHandle::size")
     virtual off_t flen() { return size(); }
+
+    /** Set blocking or non-blocking mode of the file operation like read/write.
+     *  Definition depends upon the subclass implementing FileHandle.
+     *  The default is blocking.
+     *
+     *  @param blocking true for blocking mode, false for non-blocking mode.
+     */
+    virtual int set_blocking(bool blocking) {
+        return -1;
+    }
+
+    /** Check for poll event flags
+     * The input parameter can be used or ignored - the could always return all events,
+     * or could check just the events listed in events.
+     * Call is non-blocking - returns instantaneous state of events.
+     * Whenever an event occurs, the derived class must call _poll_change().
+     * @param events bitmask of poll events we're interested in - POLLIN/POLLOUT etc.
+     * @return bitmask of poll events that have occurred.
+     */
+    virtual short poll(short events) const {
```

Comment:
Should be SHORT_MAX though, technically :)

The member poll was really intended only for the system poll().

On the system poll, I don't think a default would make sense, even if it was possible - it wouldn't be sensible to wake the poll to report a state you don't know about. So there, you always have explicit app-requested bits specified. The parameter was just to pass that required information through as a potential optimisation hint. ("Don't bother filling in these bits if it will cost you time").

We do also have to think about the EINVAL I mentioned - that poll() doesn't allow for that. Would it need a separate "not available" bit?

My feeling is I'd prefer to mark FileHandle::poll() as protected, and a friend of mbed::poll(), to make people use the real API. I don't think there's a valid application use of FileHandle::poll - it's just a non-blocking check of 1 handle, and it's less error-prone (and more portable and efficient) to just attempt non-blocking read/write calls to do that.